contracts/randdrop/src/contract.rs

```diff
@@ -0,0 +1,843 @@
+    }
+
+    if CLAIMED.has(deps.storage, &info.sender) {
+        return Err(ContractError::Claimed {});
+    }
+
+    // The contract will spend NOIS tokens on this info.sender to buy randomness.
+    // To prevent spam and users abusing the funds, we only allow addresses listed on the randdrop to call this entrypoint
+    // Sending the proof here will make sure that the sender is a potential randdrop winner
+    let merkle_root = MERKLE_ROOT.load(deps.storage)?; // This can be optimised by not loading the state twice
+    if !is_proof_valid(info.sender.clone(), amount, merkle_root, proof)? {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Get the price from the nois-proxy
+
+    let response = Response::new().add_message(WasmMsg::Execute {
+        contract_addr: config.nois_proxy.address.into_string(),
+        // GetNextRandomness requests the randomness from the proxy
+        // The job id is needed to know what randomness we are referring to upon reception in the callback.
+        msg: to_binary(&ProxyExecuteMsg::GetNextRandomness {
+            job_id: "randdrop-".to_string() + info.sender.as_str(),
+        })?,
+
+        funds: vec![config.nois_proxy.price], // Pay from the contract
+    });
+
+    // Register randdrop participant
+    let participant_data = &ParticipantData {
+        nois_randomness: None,
+        base_randdrop_amount: amount,
+        is_winner: None,
+    };
+    PARTICIPANTS.save(deps.storage, &info.sender, participant_data)?;
+
+    Ok(response)
+}
+
+pub fn execute_receive(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    callback: NoisCallback,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // callback should only be allowed to be called by the proxy contract
+    // otherwise anyone can cut the randomness workflow and cheat the randomness by sending the randomness directly to this contract
+    ensure_eq!(
+        info.sender,
+        config.nois_proxy.address,
+        ContractError::UnauthorizedReceive
+    );
+    let randomness: [u8; 32] = callback
+        .randomness
+        .to_array()
+        .map_err(|_| ContractError::InvalidRandomness)?;
+
+    // extract participant address
+    let job_id = callback.job_id;
+    let participant_address = job_id
+        .strip_prefix("randdrop-")
+        .unwrap_or_else(|| panic!("Strange, how is the job-id not prefixed with randdrop-"));
+    let participant_address = deps.api.addr_validate(participant_address)?;
+
+    // Make sure the participant is registered
+    let participant_data = PARTICIPANTS.load(deps.storage, &participant_address)?;
+    if participant_data.nois_randomness.is_some() || participant_data.is_winner.is_some() {
+        panic!("Strange, participant's randomness already received")
+    }
+    let mut msgs = Vec::<CosmosMsg>::new();
+
+    let randdrop_amount = match is_randdrop_winner(&participant_address, randomness) {
```

Comment:
Since you are matching over a bool, you can simply use an if/else

```suggestion
    let randdrop_amount = if is_randdrop_winner(&participant_address, randomness) {
```